frontend/src/components/pages/CampOverview/index.tsx

```diff
@@ -70,6 +90,98 @@ const CampOverviewPage = (): JSX.Element => {
 
   const openManageSessionsModal = () => onOpen();
 
+  const getUpdateCampSessionsRequestArray = (
+    updatedCapacities: Map<string, string>,
+  ): Array<UpdateCampSessionsRequest> => {
+    const requests: Array<UpdateCampSessionsRequest> = [];
+
+    updatedCapacities.forEach((capacity, id) => {
+      const updatedCapacityExists = capacity.trim().length > 0;
+      if (updatedCapacityExists) {
+        requests.push({ id, capacity: parseInt(capacity, 10) });
+      }
+    });
+
+    return requests;
+  };
+
+  const getManageSessionsResult = async (
+    deletedSessions: Set<string>,
+    updatedCapacities: Map<string, string>,
+  ): Promise<ManageSessionsPromisesResult> => {
```

Comment:
This is an overcomplicated approach :/ it's a small improvement that the promises are not awaited in a chain, but rather using `Promise.all`